core/trino-main/src/main/java/io/trino/execution/executor/dedicated/ThreadPerDriverTaskExecutor.java

```diff
@@ -246,4 +249,67 @@ public Set<TaskId> getStuckSplitTaskIds(Duration processingDurationThreshold, Pr
         // TODO
         return ImmutableSet.of();
     }
+
+    //
+    // STATS
+    //
+    @Managed
+    public synchronized int getTasks()
+    {
+        return tasks.size();
+    }
+
+    @Managed
+    public synchronized int getTotalRunningSplits()
+    {
+        return tasks.values().stream()
+                .mapToInt(TaskEntry::totalRunningSplits)
+                .sum();
+    }
+
+    @Managed
+    public synchronized int getTotalRunningLeafSplits()
+    {
+        return tasks.values().stream()
```

Comment:
do we need synchronization here?

Reply:
Yes, access to `tasks` needs to be synchronized (@GuardedBy("this")).